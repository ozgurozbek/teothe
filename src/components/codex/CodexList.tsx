"use client";

import Link from "next/link";
import { Card, Divider } from "antd";
import SimpleContent from "@/components/SimpleCon";
import GetCrumbs from "@/components/NavigationCrumb";

export default function CodexList({ posts }: { posts: any[] }) {
  return (
    <section>
      <GetCrumbs path={"Teothe,Codex"} />
      <Card bordered={false} className="w-full">
        <SimpleContent
          contentProps={{
            title: "Codex",
            text: [
              "Dive into rich lore, uncover immersive stories, and get exclusive gameplay tips that deepen your understanding of the realm. Whether you're looking for the latest news, detailed character arcs, or behind-the-scenes fantasy worldbuilding, the Codex is where every path begins.",
              "Stay updated with developer insights, strategy breakdowns, and curated Teothe articles; all crafted to enhance your journey through its evolving universe.",
            ],
          }}
        />
        <Divider />
        <ul>
          {posts.map(
            ({ slug, title, date, description, category, contentWarning, staffPick, duration }) => (
              <li key={slug} className="mb-4">
                <Link
                  href={`/codex/${slug}`}
                  className="text-xl text-blue-600 hover:underline"
                >
                  {title}
                </Link>
                {contentWarning && (
                  <span className="bg-[#30011a] px-2 py-1 rounded ml-2">
                    ⚠️ {contentWarning}
                  </span>
                )}
                {staffPick == "True" && (
                  <span className="bg-[#49aa19] px-2 py-1 rounded ml-2">
                    ❤️ Staff Favourite
                  </span>
                )}
                <p>{description}</p>
                <div>
                  <i className="text-sm italic">{date + ", "}</i>
                  {category && (
                    <i className="text-sm italic">{category + " category, "}</i>
                  )}
                  {duration && (<i className="text-sm italic">{duration} {duration > 1 ? "minutes" : "minute"} to read</i>)}
                </div>
              </li>
            )
          )}
        </ul>
      </Card>
    </section>
  );
}
