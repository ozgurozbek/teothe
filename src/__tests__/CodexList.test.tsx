import React from 'react';
import { render, screen } from '@testing-library/react';
import CodexList from '../components/codex/CodexList';

// Mock CodexEntry to prevent antd/next import issues
jest.mock('../components/codex/CodexEntry', () => {
  return function MockedCodexEntry({ title }: { title: string }) {
    return <li data-testid="codex-entry">{title}</li>;
  };
});

const samplePosts = [
  {
    slug: 'post-1',
    title: 'Post 1',
    date: '2025-05-10',
    description: 'Description 1',
    category: 'Category 1',
    contentWarning: 'Warning 1',
    staffPick: 'True',
    duration: 5,
  },
  {
    slug: 'post-2',
    title: 'Post 2',
    date: '2025-05-09',
    description: 'Description 2',
    category: 'Category 2',
    contentWarning: '',
    staffPick: 'False',
    duration: 1,
  },
];

describe('CodexList', () => {
  test('renders without crashing', () => {
    render(<CodexList posts={samplePosts} />);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CodexList posts={samplePosts} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders post titles using mocked CodexEntry', () => {
    render(<CodexList posts={samplePosts} />);
    expect(screen.getAllByTestId('codex-entry')).toHaveLength(2);
  });
});
