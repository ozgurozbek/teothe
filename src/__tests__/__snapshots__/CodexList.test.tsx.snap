// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodexList matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <span
      class="text-slate-400"
    >
       / Teothe
    </span>
    <span>
       / Codex
    </span>
    <div
      class="ant-card w-full css-dev-only-do-not-override-1xg9z9n"
    >
      <div
        class="ant-card-body"
      >
        <section>
          <h1
            class="ant-typography css-dev-only-do-not-override-1xg9z9n"
            data-testid="simplecon-title"
            id="codex"
          >
            Codex
          </h1>
          <div
            data-testid="simplecon-paragraph"
          >
            <p
              class="mb-2"
            >
              Dive into rich lore, uncover immersive stories, and get exclusive gameplay tips that deepen your understanding of the realm. Whether you're looking for the latest news, detailed character arcs, or behind-the-scenes fantasy worldbuilding, the Codex is where every path begins.
            </p>
            <p
              class="mb-2"
            >
              Stay updated with developer insights, strategy breakdowns, and curated Teothe articles; all crafted to enhance your journey through its evolving universe.
            </p>
          </div>
        </section>
        <div
          class="ant-divider css-dev-only-do-not-override-1xg9z9n ant-divider-horizontal"
          role="separator"
        />
        <ul>
          <li
            data-testid="codex-entry"
          >
            Post 1
          </li>
          <li
            data-testid="codex-entry"
          >
            Post 2
          </li>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
